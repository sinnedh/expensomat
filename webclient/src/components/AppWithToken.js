import React from 'react';
import { Switch, Route } from 'react-router-dom';
import ShowCalculation from '../routes/ShowCalculation';
import ExpenseDetails from '../routes/ExpenseDetails';
import ManageCalculation from '../routes/ManageCalculation';

class AppWithToken extends React.Component {
  componentDidMount() {
    this.props.onComponentDidMount()
  }

  render() {
    return (
      <Switch>
        <Route exact path="/calculation/:token/expense/:id" component={ExpenseDetails} />
        <Route exact path="/calculation/:token/manage" component={ManageCalculation} />
        <Route exact path="/calculation/:token" component={ShowCalculation} />
      </Switch>
    )
  }
}

export default AppWithToken
