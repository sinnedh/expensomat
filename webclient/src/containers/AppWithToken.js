import { connect } from 'react-redux';
import { getCalculation, getExpenses, getMembers, setToken } from '../actions'
import AppWithToken from '../components/AppWithToken';

const mapStateToProps = (state, ownProps) => ({
  token: state.getIn(['application', 'token']),
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  onComponentDidMount: () => {
    const token = ownProps.match.params.token;
    dispatch(setToken(token));
    dispatch(getCalculation(token));
    dispatch(getExpenses(token));
    dispatch(getMembers(token));
  }
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AppWithToken);
