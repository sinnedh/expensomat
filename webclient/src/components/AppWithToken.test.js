import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { MemoryRouter } from 'react-router'
import AppWithToken from './AppWithToken'

it('renders with token', () => {
  const div = document.createElement('div')
  const onComponentDidMount = jest.fn()
  ReactDOM.render(
    <MemoryRouter location="someLocation">
      <AppWithToken token={'123'} onComponentDidMount={onComponentDidMount} />
    </MemoryRouter>, div)

  expect(onComponentDidMount).toHaveBeenCalledTimes(1)
})

it('renders properly', () => {
  const wrapper = shallow(
    <MemoryRouter location="someLocation">
      <AppWithToken token={'123'} />
    </MemoryRouter>
  )

  expect(wrapper).toMatchSnapshot()
})
