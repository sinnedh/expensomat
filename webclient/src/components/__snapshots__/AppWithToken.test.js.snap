// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter
    location="someLocation"
>
    <AppWithToken
        token="123"
    />
</MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <AppWithToken
        token="123"
/>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "tqxakx",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tqxakx",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "token": "123",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <AppWithToken
          token="123"
/>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "tqxakx",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "tqxakx",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "token": "123",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
